### PR TITLE
[opt](function) opt pad function process the input const column

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before
```
mysql [test]>select count(lpad(str, 100, "abc")) from strsu;
+------------------------------+
| count(lpad(str, 100, 'abc')) |
+------------------------------+
|                     10000000 |
+------------------------------+
1 row in set (2.83 sec)

```
now
```
mysql [test]>select count(lpad(str, 100, "abc")) from strsu;
+------------------------------+
| count(lpad(str, 100, 'abc')) |
+------------------------------+
|                     10000000 |
+------------------------------+
1 row in set (1.14 sec)
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -1314,69 +1338,62 @@ class FunctionStringPad : public IFunction {
...
-                const char* pad_data =
-                        reinterpret_cast<const char*>(&padcol_chars[padcol_offsets[i - 1]]);
-
-                size_t str_char_size =
-                        simd::VStringFunctions::get_char_len(str_data, str_len, str_index);
-                size_t pad_char_size =
-                        simd::VStringFunctions::get_char_len(pad_data, pad_len, pad_index);
-
-                if (col_len_data[i] <= str_char_size) {
+                const auto str_idx = str_const ? 0 : i;
```
Comment (zclllyybb) [inline be/src/vec/functions/function_string.h:1381]: use `index_check_const`

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**